_commands/scripting/basicStdOutRedirect.md

```diff
@@ -0,0 +1,40 @@
+---
+---
+
+\> (standard output redirection operator)
+-------
+
+`command1 -[options] [arguments] > <output file name>` 
+
+The standard output redirection operator is used in bash to redirect the output of one command to a specified file.
+
+~~~ bash
+./a.out > output.txt
+~~~
+
+<!--more-->
+
+### Useful Options / Examples
+
+#### Combining with `diff`
+~~~ bash
+$ ./p3 > outfile.txt | diff outfile.txt correct.txt
+1c1
+< Hello World!
+---
+> Hello World!!
+~~~
+
```

Comment:
line 21: I don't think this line produces that output. This is the exact gotcha we put on a homework earlier in the semester. The `>` opens a new file with the name `outfile.txt` and erases the contents and THEN diffs the two files. 

Here's an example of my terminal session.

~~~bash
λ: ./main
hello world
λ: cat outfile.correct
hello world!!!
λ: ./main > outfile.out && diff outfile.out outfile.correct
1c1
< hello world
---
> hello world!!!
λ: ./main > outfile.out | diff outfile.out outfile.correct
0a1
> hello world!!!
~~~

If you want to run two commands on the same line, you can use the `&&` or `;` between commands. The only difference is that for the command after `&&` to run, the first one needs to return successfully. `;` is just like ending the line early and starting another.

~~~bash
λ: false && echo "hello"  # fails and prints nothing. AKA a 'short circuit'
λ: false; echo "hello"
hello
~~~